extractor/filesystem/language/haskell/cabal/cabal_test.go

```diff
@@ -0,0 +1,311 @@
+			path:         "software-develop/foocabal.project.freeze",
+			wantRequired: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			collector := testcollector.New()
+			var e filesystem.Extractor = cabal.New(cabal.Config{
+				Stats:            collector,
+				MaxFileSizeBytes: tt.maxFileSizeBytes,
+			})
+
+			fileSizeBytes := tt.fileSizeBytes
+			if fileSizeBytes == 0 {
+				fileSizeBytes = 1000
+			}
+
+			isRequired := e.FileRequired(simplefileapi.New(tt.path, fakefs.FakeFileInfo{
+				FileName: filepath.Base(tt.path),
+				FileMode: fs.ModePerm,
+				FileSize: fileSizeBytes,
+			}))
+			if isRequired != tt.wantRequired {
+				t.Fatalf("FileRequired(%s): got %v, want %v", tt.path, isRequired, tt.wantRequired)
+			}
+
+			gotResultMetric := collector.FileRequiredResult(tt.path)
+			if tt.wantResultMetric != "" && gotResultMetric != tt.wantResultMetric {
+				t.Errorf("FileRequired(%s) recorded result metric %v, want result metric %v", tt.path, gotResultMetric, tt.wantResultMetric)
+			}
+		})
+	}
+}
+
+func TestExtract(t *testing.T) {
+	tests := []struct {
+		name             string
+		path             string
+		cfg              cabal.Config
+		wantInventory    []*extractor.Inventory
+		wantResultMetric stats.FileExtractedResult
+	}{
+		{
+			name: "valid stack.yaml.lock file",
+			path: "testdata/valid",
+			wantInventory: []*extractor.Inventory{
+				{
+					Name:      "AC-Angle",
+					Version:   "1.0",
+					Locations: []string{"testdata/valid"},
+				},
+				{
+					Name:      "ALUT",
+					Version:   "2.4.0.3",
+					Locations: []string{"testdata/valid"},
+				},
+				{
+					Name:      "ANum",
+					Version:   "0.2.0.2",
+					Locations: []string{"testdata/valid"},
+				},
+				{
+					Name:      "Agda",
+					Version:   "2.6.4.3",
+					Locations: []string{"testdata/valid"},
+				},
+				{
+					Name:      "Allure",
+					Version:   "0.11.0.0",
+					Locations: []string{"testdata/valid"},
+				},
+			},
+			wantResultMetric: stats.FileExtractedResultSuccess,
+		},
+		{
+			name: "valid stack.yaml.lock file with package problems",
+			path: "testdata/valid_2",
+			wantInventory: []*extractor.Inventory{
+				{
+					Name:      "AC-Angle",
+					Version:   "1.0",
+					Locations: []string{"testdata/valid_2"},
+				},
+				{
+					Name:      "ANum",
+					Version:   "0.2.0.2",
+					Locations: []string{"testdata/valid_2"},
+				},
+				{
+					Name:      "Agda",
+					Version:   "2.6.4.3",
+					Locations: []string{"testdata/valid_2"},
+				},
+				{
+					Name:      "Allure",
+					Version:   "0.11.0.0",
+					Locations: []string{"testdata/valid_2"},
+				},
+			},
+			wantResultMetric: stats.FileExtractedResultSuccess,
+		},
+		{
+			name:          "invalid",
+			path:          "testdata/invalid",
+			wantInventory: []*extractor.Inventory{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fsys := scalibrfs.DirFS(".")
+
+			r, err := fsys.Open(tt.path)
```

Comment:
No need to set up files, use extracttest.GenerateScanInputMock like in https://github.com/google/osv-scalibr/pull/362